core/sector_storage/impl/remote_worker.cpp

```diff
@@ -41,15 +60,99 @@ namespace fc::sector_storage {
     return api_.Paths();
   }
 
-  RemoteWorker::RemoteWorker(io_context &context) : wsc_(context) {}
+  RemoteWorker::RemoteWorker(io_context &context)
+      : wsc_(*(worker_thread_.io)){}
+
+  struct PieceDataSender {
+    explicit PieceDataSender(io_context &io)
+        : resolver{io}, stream{net::make_strand(io)} {}
+    PieceDataSender(const PieceDataSender &) = delete;
+    PieceDataSender(PieceDataSender &&) = delete;
+    ~PieceDataSender() {
+      boost::system::error_code ec;
+      stream.socket().shutdown(tcp::socket::shutdown_both, ec);
+    }
+    PieceDataSender &operator=(const PieceDataSender &) = delete;
+    PieceDataSender &operator=(PieceDataSender &&) = delete;
+
+    static void send(
+        int fd,
+        io_context &io,
+        const std::string &host,
+        const std::string &port,
+        const std::string &target,
+        const std::uint64_t piece_size,
+        const std::function<void(outcome::result<std::string>)> &cb) {
+      auto s{std::make_shared<PieceDataSender>(io)};
```

Comment:
Use names that describe the purpose or intent of the object. Do not worry about saving horizontal space as it is far more important to make your code immediately understandable by a new reader.
```suggestion
      auto sender{std::make_shared<PieceDataSender>(io)};
```

Reply:
done